### PR TITLE
fix script for non-cuda devices

### PR DESCRIPTION
**Pull request by adamskrodzki**

### CODE REVIEW

File: train_gpt2.py [hunk 1]
```diff
@@ -505,7 +505,8 @@ def get_lr(it):
     for param_group in optimizer.param_groups:
         param_group['lr'] = lr
     optimizer.step()
-    torch.cuda.synchronize() # wait for the GPU to finish work
+    if(device.startswith("cuda")):
```
Comment (karpathy) [inline train_gpt2.py:508]: not proper python, do not use brackets.
also, maybe use device_type == 'cuda' instead?

**Comment by karpathy** [comment]
fixed

File: train_gpt2.py (hunk 1 above)
Comment (adamskrodzki) [reply train_gpt2.py:508]: keeping consistent with:

https://github.com/karpathy/build-nanogpt/blob/6104ab1b53920f6e2159749676073ff7d815c1fa/train_gpt2.py#L316

Comment (adamskrodzki) [reply train_gpt2.py:508]: on bracket's I'm not really python dev to argue, but this code works instead of blowing up on cpu - profit ;)

Comment (adamskrodzki) [reply train_gpt2.py:508]: Ok I see you fixed it yourself, so thank

**Pull request closed without merging**